crates/wasmtime/src/func.rs

```diff
@@ -1100,65 +971,54 @@ impl Func {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn typed<Params, Results>(&self) -> Result<&TypedFunc<Params, Results>>
+    pub fn typed<Params, Results, S>(&self, store: S) -> Result<TypedFunc<Params, Results>>
     where
         Params: WasmParams,
         Results: WasmResults,
+        S: AsContext,
     {
         // First type-check that the params/results are all valid...
-        let ty = self.ty();
+        let ty = self.ty(store);
         Params::typecheck(ty.params()).context("type mismatch with parameters")?;
         Results::typecheck(ty.results()).context("type mismatch with results")?;
 
         // ... then we can construct the typed version of this function
         // (unsafely), which should be safe since we just did the type check above.
-        unsafe { Ok(self.typed_unchecked::<Params, Results>()) }
-    }
-
-    /// An unchecked version of [`Func::typed`] which does not perform a
-    /// typecheck and simply assumes that the type declared here matches the
-    /// type of this function.
-    ///
-    /// The semantics of this function are the same as [`Func::typed`] except
-    /// that no error is returned because no typechecking is done.
-    ///
-    /// # Unsafety
-    ///
-    /// This function only safe to call if `typed` would otherwise return `Ok`
-    /// for the same `Params` and `Results` specified. If `typed` would return
-    /// an error then the returned `TypedFunc` is memory unsafe to invoke.
-    pub unsafe fn typed_unchecked<Params, Results>(&self) -> &TypedFunc<Params, Results>
```

Comment:
I removed this in favor of `TypedFunc::new_unchecked` to make its presence a little less prominent.